src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -515,6 +517,113 @@ spanning_coe_le_of_le (delete_edges_le s)
 
 end delete_edges
 
+/-! ## Induced subgraphs and vertex deletion -/
+
+/- Given a subgraph, we can change its vertex set while removing any invalid edges, and
+this leads to induced subgraphs and vertex deletion. See also `simple_graph.induce` for the
+`simple_graph` version, which, unlike for subgraphs, results in a graph with a different type. -/
+
+/-- The induced subgraph of a subgraph. The expectation is that `s ⊆ G'.verts` for the usual
+notion of an induced subgraph, but, in general, `s` is taken to be the new vertex set and edges
+are induced from the subgraph `G'`. -/
+@[simps]
+def induce (G' : G.subgraph) (s : set V) : G.subgraph :=
+{ verts := s,
+  adj := λ u v, u ∈ s ∧ v ∈ s ∧ G'.adj u v,
+  adj_sub := λ u v, by { rintro ⟨-, -, ha⟩, exact G'.adj_sub ha },
+  edge_vert := λ u v, by { rintro ⟨h, -, -⟩, exact h } }
+
+lemma _root_.simple_graph.induce_eq_coe_induce_top (s : set V) :
+  G.induce s = ((⊤ : G.subgraph).induce s).coe :=
+by { ext v w, simp }
+
+section induce
+variables {G' G'' : G.subgraph} {s s' : set V}
+
+lemma induce_mono (hg : G' ≤ G'') (hs : s ⊆ s') : G'.induce s ≤ G''.induce s' :=
```

Comment:
In such cases, the one-sided monotonicity lemmas empirically seem the most useful, because it rarely happens that you want to change both sides simultaneously. It's to the point where I do not advise to have the two-sided lemma anymore, but keeping it is fine, of course.

Reply:
I added convenience one-sided lemmas. It's easier to prove the "bifunctor" version directly in this case.